ddl/ddl_api.go

```diff
@@ -3266,3 +3307,19 @@ func buildPartitionInfo(meta *model.TableInfo, d *ddl, spec *ast.AlterTableSpec)
 	}
 	return part, nil
 }
+
+// extractCollateFromOption take collates(may multiple) in option into consideration
+// when handle charset and collate of a column, rather than handling it separately.
+func extractCollateFromOption(def *ast.ColumnDef) []string {
+	specifiedCollates := make([]string, 0, 0)
+	for i := 0; i < len(def.Options); i++ {
+		op := def.Options[i]
+		if op.Tp == ast.ColumnOptionCollate {
+			specifiedCollates = append(specifiedCollates, op.StrValue)
+			def.Options = append(def.Options[:i], def.Options[i+1:]...)
```

Comment:
What will happen if we don't delete the `ColumnOptionCollate` option in ColumnDef?

Reply:
Here if we don't remove it in `Option`, it is ok for the subsequent logic, not a big deal. 
But for `ast.Restore` func, it will cause duplicate collate in restored SQL string.

Reply:
> What will happen if we don't delete the `ColumnOptionCollate` option in ColumnDef?

Here if we don't remove it in Option, it is ok for the subsequent logic, not a big deal.
But for ast.Restore func, it will cause duplicate collate in restored SQL string.